sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/TablesExtensionConfigProvider.cs

```diff
@@ -165,6 +189,51 @@ await foreach (var entity in entities)
             return entityArray;
         }
 
+        private async Task<ParameterBindingData[]> CreateEnumerable(TableAttribute attribute, ValueBindingContext context)
+        {
+            var table = GetTable(attribute);
+
+            string filter = attribute.Filter;
+            if (!string.IsNullOrEmpty(attribute.PartitionKey))
+            {
+                var partitionKeyPredicate = TableClient.CreateQueryFilter($"PartitionKey eq {attribute.PartitionKey}");
+                if (!string.IsNullOrEmpty(filter))
+                {
+                    filter = $"{partitionKeyPredicate} and {filter}";
+                }
+                else
+                {
+                    filter = partitionKeyPredicate;
+                }
+            }
+
+            int? maxPerPage = null;
+            if (attribute.Take > 0)
+            {
+                maxPerPage = attribute.Take;
+            }
+
+            int countRemaining = attribute.Take;
+
+            List<ParameterBindingData> bindingDataContent = new List<ParameterBindingData>();
+            var entities = table.QueryAsync<TableEntity>(
+                filter: filter,
+                maxPerPage: maxPerPage).ConfigureAwait(false);
+
+            await foreach (var entity in entities)
+            {
+                countRemaining--;
```

Comment:
What will happen if `Take` is not specified? Will Attribute.Take be zero or something else or is it mandatory to specify Take?

And if Attribute.Take is 0 then do we want to return empty array from this method?

Reply:
If take is not specified it won't limit the number of entries that are returned to the user. I tried out the in-proc scenario of Take is 0 and it seems to ignore the value of take and just returns all the values. I did the same thing for the out of proc scenario

Reply:
Added a unit test for this scenario